configs/field-types/media.jsx

```diff
@@ -0,0 +1,158 @@
+    },
+  },
+  inputWrapper: {
+    position: 'relative',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    flexDirection: 'column',
+    height: '100%',
+    width: '100%',
+    border: '1px dashed silver',
+  },
+  input: {
+    position: 'absolute',
+    height: '100%',
+    width: '100%',
+    opacity: 0,
+  },
+  removeIcon: {
+    position: 'absolute',
+    right: 5,
+    cursor: 'pointer',
+  },
+  downloadIcon: {
+    height: 30,
+    width: 30,
+  },
+  dragInfoLabel: {
+    fontSize: 12,
+    fontWeight: 700,
+    color: 'black',
+  },
+  clickInfoLabel: {
+    fontSize: 10,
+    color: colors.grey[500],
+    paddingBottom: 10,
+  },
+});
+
+export const Renderer = asField(({ fieldState, fieldApi, label, ...props }) => {
+  useEffect(() => {
+    loadMedia(value).then(res => res && setMediaUrl(`/static/${value}`));
+  }, [value])
+  const { value } = fieldState;
+  const { setValue, setTouched } = fieldApi;
+  const { onChange, onBlur, initialValue, forwardedRef, ...rest } = props;
+  const [mediaUrl, setMediaUrl] = useState(null);
+  const [mediaHover, setMediaHover] = useState(false);
+  const classes = useStyles();
+
+  return (
+    <div className={classes.wrapper}>
+      {label && <InputLabel>{S(label).capitalize().s}</InputLabel>}
+      {mediaUrl === null ? (
+        <div
+          className={classes.dropZoneWrapper}
+          onClick={() => Router.push('/admin/media')}
+          onKeyUp={() => {}}
+          role="button"
+          tabIndex="0"
+        >
+          <div className={classes.inputWrapper}>
+            <input
+              {...rest}
+              ref={forwardedRef}
+              type="file"
+              multiple={true}
+              onClick={e => e.preventDefault()}
+              onChange={e => handleChange(e, setValue, setMediaUrl)}
+              onBlur={e => {
+                setTouched(true);
+                if (onBlur) {
+                  onBlur(e);
+                }
+              }}
+              className={classes.input}
+            />
+            <div className={classes.dragInfoLabel}>DRAG AND DROP A FILE</div>
+            <div className={classes.clickInfoLabel}>(OR CLICK TO BROWSE)</div>
+            <DownloadIcon className={classes.downloadIcon} />
+          </div>
+        </div>
+      ) : (
+        <div
+          className={classes.previewWrapper}
+          onKeyUp={() => {}}
+          role="button"
+          tabIndex="0"
+          onMouseEnter={() => setMediaHover(true)}
+          onMouseLeave={() => setMediaHover(false)}
+        >
+          <img src={mediaUrl} className={classes.imgWrapper} alt="" />
+          {mediaHover && (
+            <RemoveIcon
+              onClick={() => handleRemoveMedia(setMediaUrl, setValue)}
+              className={classes.removeIcon}
+            />
+          )}
+        </div>
+      )}
+    </div>
+  );
+});
+
+const handleChange = (e, setValue, setMediaUrl) => {
```

Comment:
lets use `useCallback` here for such cases
https://reactjs.org/docs/hooks-reference.html#usecallback